github/WorkflowJob.py

```diff
@@ -0,0 +1,201 @@
+################################################################################
+
+import github.GithubObject
+import github.WorkflowStep
+
+
+class WorkflowJob(github.GithubObject.CompletableGithubObject):
+    """
+    This class represents Workflow Jobs. The reference can be found here https://docs.github.com/en/rest/reference/actions#workflow-jobs
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"id": self._id.value, "url": self._url.value})
+
+    @property
+    def check_run_url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._check_run_url)
+        return self._check_run_url.value
+
+    @property
+    def completed_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._completed_at)
+        return self._completed_at.value
+
+    @property
+    def conclusion(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._conclusion)
+        return self._conclusion.value
+
+    @property
+    def head_sha(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._head_sha)
+        return self._head_sha.value
+
+    @property
+    def html_url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._html_url)
+        return self._html_url.value
+
+    @property
+    def id(self):
+        """
+        :type: int
+        """
+        self._completeIfNotSet(self._id)
+        return self._id.value
+
+    @property
+    def name(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._name)
+        return self._name.value
+
+    @property
+    def node_id(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._node_id)
+        return self._node_id.value
+
+    @property
+    def run_id(self):
+        """
+        :type: integer
+        """
+        self._completeIfNotSet(self._run_id)
+        return self._run_id.value
+
+    @property
+    def run_url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._run_url)
+        return self._run_url.value
+
+    @property
+    def started_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._started_at)
+        return self._started_at.value
+
+    @property
+    def status(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._status)
+        return self._status.value
+
+    @property
+    def steps(self):
+        """
+        :type: list of github.WorkflowStep.WorkflowStep
+        """
+        self._completeIfNotSet(self._steps)
+        return self._steps.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._url)
+        return self._url.value
+
+    def logs_url(self):
```

Comment:
How come `WorflowRun.logs_url` is an ordinary string attribute?

https://github.com/PyGithub/PyGithub/blob/554b2b28db8e3c06915406cb10e39aceb38d16d7/github/WorkflowRun.py#L167-L173

What is different in `WorkflowJob`?

Reply:
I think `logs_url` is a field that has been added to the response since I originally wrote the code. I can see the captured response payload inside `tests/ReplayData/WorkflowRun.test_jobs.txt` doesn't have it, but the new one inside `tests/ReplayData/WorkflowJob.setUp.txt` does.

It can probably be changed to work the same way as `WorkflowRun` does.

Reply:
Fix added as a suggestion.